### DISCUSSION TITLE
Ways to redirect traffic from firebase iOS SDK to localhost for XCUITest

### DISCUSSION THREAD

**Discussion by jesus-mg-ios** [firebase/firebase-ios-sdk#10256] [Q&A]
Hello community:

I'm planning use firebase in a new personal project, but I've noticed that it's sometimes too slow for UI Testing (I could also see some posts on StackOverflow talking about it), I mean for example repeatedly calling remote config (delays a test from 3 to 10 seconds at worst) might be too slow. With few tests, the times are fine, but for a large test batch, the test time increases rapidly.

I don't know if it is possible to redirect SDK traffic to a localhost server and how I can do it. Using the Firebase emulator is not an alternative for me, because I don't want to couple the app to this emulator (more dependencies). Including compiler "if else"s in the target application to test, are the last alternative I have in mind.

Thanks for everything

**Comment by paulb777** [comment]
See the FakeConsole used for RemoteConfig integration testing at https://github.com/firebase/firebase-ios-sdk/tree/master/FirebaseRemoteConfigSwift/Tests
